sams/sampler/NvidiaSMI.py

```diff
@@ -166,12 +166,15 @@ def do_sample(self):
 
     def sample(self):
         logger.debug("sample()")
+        self._most_recent_sample = []
         while not self.smi.queue.empty():
             data = self.smi.queue.get()
             logger.debug(data)
             index = data["index"]
             del data["index"]
-            self.store({index: data})
+            entry = {index: data}
+            self._most_recent_sample.append(self._storage_wrapping(entry))
+            self.store(entry)
```

Comment:
```suggestion
            most_recent_sample.append(self._storage_wrapping(entry))
            self.store(entry)
        self._most_recent_sample = most_recent_sample
```

... and then another here. This should make all edits to `sefl_most_recent_sample` atomic.